src/qt/window.cpp

```diff
@@ -1496,7 +1496,11 @@ bool wxWindowQt::QtHandleCloseEvent ( QWidget *handler, QCloseEvent *WXUNUSED( e
 bool wxWindowQt::QtHandleContextMenuEvent ( QWidget *WXUNUSED( handler ), QContextMenuEvent *event )
 {
     wxContextMenuEvent e( wxEVT_CONTEXT_MENU, GetId() );
-    e.SetPosition( wxQtConvertPoint( event->globalPos() ) );
+    e.SetPosition(
+        event->reason() != QContextMenuEvent::Keyboard ?
+        wxQtConvertPoint( event->globalPos() ) :
+        wxDefaultPosition
```

Comment:
"Default position" for showing context menu from kb looks fine.
As a personal taste, I'd probably have put the positive case first which should be easier to read IMO.